pkg/cmd/clustertask/create.go

```diff
@@ -0,0 +1,118 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clustertask
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	ctactions "github.com/tektoncd/cli/pkg/clustertask"
+	"github.com/tektoncd/cli/pkg/formatted"
+	"github.com/tektoncd/cli/pkg/task"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type createOptions struct {
+	From string
+}
+
+func createCommand(p cli.Params) *cobra.Command {
+	opts := &createOptions{}
+	eg := `Create a ClusterTask from Task 'foo' present in namespace 'ns':
+	tkn clustertask create --from foo
+or
+	tkn clustertask create foobar --from=foo`
+
+	c := &cobra.Command{
+		Use:               "create",
+		ValidArgsFunction: formatted.ParentCompletion,
+		Short:             "Create a ClusterTask from Task",
+		Example:           eg,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			s := &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+			if opts.From == "" {
+				return errors.New("--from flag not passed")
+			}
+
+			if len(args) != 0 {
+				return createClusterTaskFromTask(s, p, args[0], opts.From)
+			}
+			return createClusterTaskFromTask(s, p, opts.From, opts.From)
```

Comment:
Why the same argument passed twice?

Reply:
as there can be two scenarios
`tkn create task foobar --from foo` and `tkn create task --from foo` so in the latter case the name of the task would be the same as that of clustertask. Probably I'll try to improve.